src/rpc/mod.rs

```diff
@@ -181,17 +181,23 @@ impl ConnectionSender {
     }
 
     #[allow(dead_code)]
-    /// close the connection
+    /// Close the connection
+    /// Requests in requests map will be cancelled when the writer closed
     async fn close(&mut self) -> Result<()> {
         self.request(ObTablePacket::ClosePoison).await?;
-        self.shutdown()
+        if let Some(writer) = mem::take(&mut self.writer) {
+            writer.await??
+        }
+        self.sender.closed().await;
+        Ok(())
     }
 
     /// Shutdown the sender without closing remote
+    /// Requests in the requests map will be not be cancelled
     fn shutdown(&mut self) -> Result<()> {
-        let writer = mem::replace(&mut self.writer, None);
-        let drop_helper = AbortOnDropMany(vec![writer.unwrap()]);
-        drop(drop_helper);
+        if let Some(writer) = mem::take(&mut self.writer) {
+            let _drop_helper = AbortOnDropMany(vec![writer]);
```

Comment:
`writer.abort()` is enough to avoid one more memory allocation for AbortOnDropMany.

Reply:
Fixed.